app/javascript/components/lease_ups/utils/leaseUpRequestUtils.js

```diff
@@ -25,11 +27,16 @@ export const getApplications = async (listingId, page, filters) => {
   }
   return apiService
     .fetchLeaseUpApplications(listingId, page, { filters })
-    .then(({ records, pages }) => {
-      const preferences = map(records, buildLeaseUpAppPrefModel)
-      const layeredPreferences = addLayeredValidation(preferences)
+    .then(({ records, pages, listing_type }) => {
+      let apps
+      if (listing_type === LISTING_TYPE_FIRST_COME_FIRST_SERVED) {
+        apps = map(records, buildLeaseUpAppFirstComeFirstServedModel)
+      } else {
+        const preferences = map(records, buildLeaseUpAppPrefModel)
+        apps = addLayeredValidation(preferences)
+      }
```

Comment:
Good q! In preference_service, we are querying preferences, so when a user searches, their searches and filters are supposed to apply to the application, but in the backend, we still do the query on preferences, so we need index into the associated application and then filter/search on those

In lease_up_application_service, we are now querying applications directly (because the fcfs listings don't have preferences), so we don't have to index into the application object from the preference anymore - we are already starting with application

Old query:
Grab all preference objects, index into associated application, apply searches and filters

New query:
Grab all applications, apply searches and filters

Reply:
omg great call because in writing this test I realized some data missing from the new `leaseUpAppFirstComeFirstServedModel.js` - yay for test driven development!